### PR TITLE
Refactor rails api

### PR DESCRIPTION
**Pull request by warrenshen**

### CODE REVIEW

**Comment by warrenshen** [comment]
@Byronium This is ready!

File: phc-rails/Gemfile [hunk 1]
```diff
@@ -1,17 +1,14 @@
 source "https://rubygems.org"
 
-# Living life on the edge with the most recent version of ruby
-# ruby "2.1.3"
-
 # Bundle edge Rails instead: gem "rails", github: "rails/rails"
-gem "rails", "4.1.5"
+gem "rails"
```
Comment (chenhoward) [inline phc-rails/Gemfile]: I think you should specify a version number

File: phc-rails/app/controllers/api/v1/event_registrations_controller.rb [hunk 2]
```diff
@@ -46,49 +47,45 @@ def get_applied
...
-      raise "Service status is not known. This should not happen!"
+      api_message_response(500)
     end
   end
 
   def update_feedback
     @event_registration = EventRegistration.find_by(Number__c: params[:Number__c])
     if @event_registration.update(event_registration_params)
-      render :json => { status: 201, message: "Good job Shimmy" }
+      api_message_response(201, "Good job Shimmy")
```
Comment (chenhoward) [inline phc-rails/app/controllers/api/v1/event_registrations_controller.rb]: LOL is this our actual message doe

File: phc-rails/app/controllers/api/v1/event_registrations_controller.rb [hunk 3]
```diff
@@ -31,13 +31,14 @@ def create
       (event_reg.services ||= []) << Service.new(name: service, status:status)
     end
 
-    render :json => { status: (event_reg.save ? "Success" : "Failure") }
+    status = event_reg.save ? "Success" : "Failure"
```
Comment (chenhoward) [inline phc-rails/app/controllers/api/v1/event_registrations_controller.rb]: I actually find the ternary not railsy

File: phc-rails/app/models/service.rb [hunk 4]
```diff
@@ -6,30 +6,21 @@
...
 
-  def self.DROPIN
-    "Drop In"
-  end
+  enum status: [:unspecified, :applied, :drop_in, :received]
 
-  def self.RECIEVED
-    "Received"
-  end
+  ##################################################
```
Comment (chenhoward) [inline phc-rails/app/models/service.rb]: Oh shit the section comment blocks doe

File: phc-rails/app/controllers/static_pages_controller.rb [hunk 5]
```diff
@@ -1,2 +1,3 @@
 class StaticPagesController < ApplicationController
+
```
Comment (chenhoward) [inline phc-rails/app/controllers/static_pages_controller.rb]: Ag

**Comment by chenhoward** [comment]
Check out my comments and lmk what you think!

File: phc-rails/app/controllers/api/v1/event_registrations_controller.rb (hunk 2 above)
Comment (warrenshen) [inline phc-rails/app/controllers/api/v1/event_registrations_controller.rb]: Yes sir for now it is!

File: phc-rails/app/controllers/api/v1/event_registrations_controller.rb (hunk 3 above)
Comment (warrenshen) [inline phc-rails/app/controllers/api/v1/event_registrations_controller.rb]: You're going to have to talk to @altonzheng about this (I can't take credit for this code).

File: phc-rails/Gemfile (hunk 1 above)
Comment (warrenshen) [inline phc-rails/Gemfile]: Swell.

**Comment by warrenshen** [comment]
@chenhoward thanks for the code review! @Byronium ready again.

File: phc-rails/app/controllers/api/v1/event_registrations_controller.rb (hunk 3 above)
Comment (Byronium) [inline phc-rails/app/controllers/api/v1/event_registrations_controller.rb]: howard what would you suggest instead?

File: phc-rails/app/controllers/api/v1/accounts_controller.rb [hunk 6]
```diff
@@ -1,11 +1,14 @@
...
   before_action :verify_security_token
 
   def show
-    sf_id = params[:sf_id]
-    respond_with Account.find_by sf_id: sf_id
+    account = Account.find_by(sf_id: params[:sf_id])
+    if !account.nil?
+      render json: account
+    else
+      api_message_response(401, "Account with that id does not exist.")
```
Comment (Byronium) [inline phc-rails/app/controllers/api/v1/accounts_controller.rb]: what gem is api_message_response part of?

File: phc-rails/app/controllers/api/v1/event_registrations_controller.rb (hunk 2 above)
Comment (Byronium) [inline phc-rails/app/controllers/api/v1/event_registrations_controller.rb]: we should actually update the event_registration's experiences, services needed, and feedback now that I created it in Salesforce.

**Comment by Byronium** [comment]
@warrenshen i just had 3 quick questions/concerns! could you address those? then i'll be ready to merge

File: phc-rails/app/controllers/api/v1/accounts_controller.rb (hunk 6 above)
Comment (warrenshen) [inline phc-rails/app/controllers/api/v1/accounts_controller.rb]: api_message_response is a helper method I defined in the `ApplicaitionController`. It's inherited by all other controllers and helps reduce repetition of code.

See https://github.com/calblueprint/phc/pull/338/files#diff-a0c74587805b9556430a52e4c902f2ccR33.

File: phc-rails/app/controllers/api/v1/event_registrations_controller.rb (hunk 3 above)
Comment (warrenshen) [inline phc-rails/app/controllers/api/v1/event_registrations_controller.rb]: I think we can just leave this as it is for now!

File: phc-rails/app/controllers/api/v1/event_registrations_controller.rb (hunk 2 above)
Comment (warrenshen) [inline phc-rails/app/controllers/api/v1/event_registrations_controller.rb]: There isn't a feedback model yet! I can make that on Monday and fully set this action up.

**Comment by warrenshen** [comment]
@Byronium responded and ready!

**Comment by warrenshen** [comment]
Deleting this branch since it's in!

**Pull request merged**